### PR TITLE
Simplify EndpointManager class hierarchy

### PR DESCRIPTION
**Pull request by pveentjer**
Removed: 
- MemberViewUnifiedEndpointManager
- ClientViewUnifiedEndpoint 
- TextViewUnifiedEndpoint 

These classes don't don't do any real work, they forward to the TcpIpUnifiedEndpoint. Only the TextViewViewUnifiedEndpoint requires ability to extract the number of REST connections. This has been added by adding a method:

 public int getConnectionCount(String connectionType) 

To the TcpIpEndpointManager and calling this method from the HttpGetCommandProcessor.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/nio/tcp/TcpIpUnifiedEndpointManager.java [hunk 1]
```diff
@@ -25,18 +25,13 @@
...
-
 import static com.hazelcast.internal.metrics.MetricDescriptorConstants.TCP_METRIC_UNIFIED_ENDPOINT_MANAGER_CLIENT_COUNT;
 import static com.hazelcast.internal.metrics.MetricDescriptorConstants.TCP_METRIC_UNIFIED_ENDPOINT_MANAGER_TEXT_COUNT;
 import static com.hazelcast.internal.metrics.ProbeLevel.MANDATORY;
 import static com.hazelcast.internal.nio.ConnectionType.MEMCACHE_CLIENT;
 import static com.hazelcast.internal.nio.ConnectionType.REST_CLIENT;
 
-class TcpIpUnifiedEndpointManager
-        extends TcpIpEndpointManager {
+class TcpIpUnifiedEndpointManager extends TcpIpEndpointManager {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/nio/tcp/TcpIpUnifiedEndpointManager.java]: Probably we can get rid of this class as well. The only thing it adds are some metrics; these could be add conditionally in the TcpIpEndpointManager,

File: hazelcast/src/test/java/com/hazelcast/internal/nio/tcp/TcpIpEndpointManager_ConnectionListenerTest.java [hunk 2]
```diff
@@ -68,26 +60,21 @@ public void whenConnectionDestroyed() {
...
-        networkingServiceA.getEndpointManager(MEMBER).addConnectionListener(listener);
-
-        networkingServiceA.shutdown();
-
-        final MemberViewUnifiedEndpointManager endpointManager = (MemberViewUnifiedEndpointManager) networkingServiceA
-                .getEndpointManager(EndpointQualifier.MEMBER);
-
-        assertEquals(0, endpointManager.getConnectionListenersCount());
-    }
+//    @Test
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/internal/nio/tcp/TcpIpEndpointManager_ConnectionListenerTest.java]: Need to fix this.

**Comment by pveentjer** [comment]
Probably EndpointManager should be renamed to ConnectionManager since it doesn't manage endpoints but connections.

File: hazelcast/src/main/java/com/hazelcast/internal/nio/EndpointManager.java [hunk 3]
```diff
@@ -28,6 +28,8 @@
 public interface EndpointManager<T extends Connection>
         extends ConnectionListenable, Consumer<Packet> {
 
+    int getConnectionCount(String connectionType);
```
[the remark this answers is not in the document]
Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/internal/nio/EndpointManager.java]: I disagree. Connection type is a fundamental concept within the Connection API. Having a method that utilizes this part of the API therefor makes sense. 

Also we only nee a single TcpIpEndpointManager classes which can contain a mix of different types of connections.

File: hazelcast/src/main/java/com/hazelcast/internal/nio/tcp/TcpIpEndpointManager.java [hunk 4]
```diff
@@ -152,6 +152,14 @@
         }
     }
 
+    public int getConnectionCount(String connectionType) {
```
[the remark this answers is not in the document]
Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/internal/nio/tcp/TcpIpEndpointManager.java]: This instance is being used for multiple situations. It is used in the 'unified' and the 'seperated' configuration. This method is needed for the unified approach.

File: hazelcast/src/main/java/com/hazelcast/internal/nio/EndpointManager.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (tkountis) [reply hazelcast/src/main/java/com/hazelcast/internal/nio/EndpointManager.java]: Its ok to disagree :-) its only healthy.
However, the design was made around Endpoints, (Endpoint Config, Endpoint Qualifier, Endpoint Manager). A single Endpoint Manager, manages a single socket = single connection type, this is all the advanced networking.
There is one exception to that, which is the Unified socket, which so far was hidden in the details of the Unified implementation. Now with this change, we break the contract of the single socket API.

**Comment by Holmistr** [comment]
run-lab-run

**Comment by hz-highfive** [comment]
The job `hazelcast` of your PR failed ([log](sss), [artifacts](aaa)). 
Through arcane magic we have determined that the following fragments from the build log may contain 
information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
bbb
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/6208/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/6208/Hazelcast-pr-builder-6208.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain 
information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] 
Tests run: 24273, Failures: 0, Errors: 0, Skipped: 995
</details>
</pre>

**Comment by Holmistr** [comment]
run-lab-run

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/6210/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/6210/Hazelcast-pr-builder-6210.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain 
information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>

</details>
</pre>

**Comment by Holmistr** [comment]
run-lab-run

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/6214/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/6214/Hazelcast-pr-builder-6214.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain 
information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>

</details>
</pre>

**Comment by Holmistr** [comment]
run-lab-run

**Comment by Holmistr** [comment]
run-lab-run

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/5/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/5/Hazelcast-pr-builder-5.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain 
information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>

</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/6/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/6/Hazelcast-pr-builder-6.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain 
information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TcpIpEndpointManagerMetricsTest.testUnifiedEndpointManagerMetricsCollectedOnce:55->verifyCollectedOnce:64 expected:<1> but was:<5>
[INFO] 
[ERROR] Tests run: 3854, Failures: 1, Errors: 0, Skipped: 14
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/9/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/9/Hazelcast-pr-builder-9.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain 
information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TcpIpEndpointManagerMetricsTest.testUnifiedEndpointManagerMetricsCollectedOnce:55->verifyCollectedOnce:64 expected:<1> but was:<5>
[INFO] 
[ERROR] Tests run: 3854, Failures: 1, Errors: 0, Skipped: 14
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Comment by Holmistr** [comment]
run-lab-run

**Comment by Holmistr** [comment]
run-lab-run

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/13/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/13/Hazelcast-pr-builder-13.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain 
information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TcpIpEndpointManagerMetricsTest.testUnifiedEndpointManagerMetricsCollectedOnce:55->verifyCollectedOnce:64 expected:<1> but was:<5>
[INFO] 
[ERROR] Tests run: 3854, Failures: 1, Errors: 0, Skipped: 14
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Pull request closed without merging**